### PR TITLE
Exchange: Poloniex fixes

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

Fixes panic in websocket as the endpoint sends different types.
Fixes nonce generation issue as it requires unixnano timestamp. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Travis with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/poloniex/poloniex.go [hunk 1]
```diff
@@ -785,7 +785,7 @@ func (p *Poloniex) SendAuthenticatedHTTPRequest(method, endpoint string, values
 	headers := make(map[string]string)
 	headers["Content-Type"] = "application/x-www-form-urlencoded"
 	headers["Key"] = p.API.Credentials.Key
-	values.Set("nonce", p.Requester.GetNonce(true).String())
+	values.Set("nonce", strconv.FormatInt(time.Now().UnixNano(), 10))
```
Comment (gloriousCode) [inline exchanges/poloniex/poloniex.go:788]: Would you be able to elaborate on this `Fixes nonce generation issue as it requires unixnano timestamp.`?
`p.Requester.GetNonce(true)` is a unixnano timestamp.

Comment (shazbert) [reply exchanges/poloniex/poloniex.go:788]: Doesn't like the timestamp to be incremented needs to be within a certain threshold per call. Don't have funds on the exchange, need to do a quick limit order submissions to test if we need to add in specific padding between calls now that I think about it.

Comment (gloriousCode) [reply exchanges/poloniex/poloniex.go:788]: This does not seem to match their documentation:
>Additionally, all queries must include a "nonce" POST parameter. The nonce parameter is an integer which must always be greater than the previous nonce used and does not need to increase by one. Using the epoch in milliseconds is an easy choice here but be careful about time synchronization if using the same API key across multiple servers.
Did you have multiple instances running by any chance?

Comment (shazbert) [reply exchanges/poloniex/poloniex.go:788]: image

Tested with for loop deploying limit order. The update seems fine. Just need somebody else to confirm.

Comment (xtda) [reply exchanges/poloniex/poloniex.go:788]: > Would you be able to elaborate on this `Fixes nonce generation issue as it requires unixnano timestamp.`?
> `p.Requester.GetNonce(true)` is a unixnano timestamp.

GetNonce() is technically only the current time in UnixNano the first request additional requests by the same requestor instance just increment by one this has thrown exchanges off in the past as well

**Pull request merged**